File: app/src/main/java/com/example/asthmaapplication/main/mainpage/MainActivity.java
package com.example.asthmaapplication.main.mainpage;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.example.asthmaapplication.R;
import com.example.asthmaapplication.main.common.BaseActivity;
import com.example.asthmaapplication.main.utils.ActivityUtils;

import javax.annotation.Nullable;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        MainFragment fragment = new MainFragment();
        ActivityUtils.addFragment(getSupportFragmentManager(), fragment, R.id.fragment_container);

    }

}

File: app/src/main/java/com/example/asthmaapplication/main/mainpage/MainActivity.kt
package com.example.asthmaapplication.main.mainpage

import dagger.hilt.android.AndroidEntryPoint
import com.example.asthmaapplication.main.common.BaseActivity
import android.os.Bundle
import com.example.asthmaapplication.main.utils.ActivityUtils
import com.example.inventory.R

@AndroidEntryPoint
class MainActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        val fragment = MainFragment()
        ActivityUtils.addFragment(supportFragmentManager, fragment, R.id.fragment_container)
    }
}